eth/filters/filter_system.go

```diff
@@ -393,23 +397,33 @@ func (es *EventSystem) lightFilterLogs(header *types.Header, addresses []common.
 }
 
 // eventLoop (un)installs filters and processes mux events.
-func (es *EventSystem) eventLoop() {
+func (es *EventSystem) eventLoop(initWaitGroup *sync.WaitGroup) {
```

Comment:
it looks like that async part can be spawned after all required variables are initialized

Reply:
Wouldn't the race condition checker scream at us in that case?

(I assume you mean that we spawn a goroutine inside `eventLoop` after all the init work, instead of having the whole `eventLoop` inside a goroutine, like now).

Reply:
i think it is safe to do so in go, can this case be reduced to?

```
func main() {
	a := map[string]string{}
	a["c"] = "c"
	go func() {
		a["c"] = "b"
	}()
	time.Sleep(100 * time.Millisecond)
}
```

Reply:
Maybe you're right... But this is also a temporary patch, this code is heavily redesigned in geth 1.8 and doesn't have this flaw anymore :)

Reply:
sure, if you feel that it is enough i am ok with it